LibreNMS/service.py

```diff
@@ -329,16 +391,16 @@ def fetch_immediate_device_list(self):
 
             result = self._db.query('''SELECT `device_id`,
                   `poller_group`,
-                  COALESCE(`last_polled` <= DATE_ADD(DATE_ADD(NOW(), INTERVAL -%s SECOND), INTERVAL `last_polled_timetaken` SECOND), 1) AS `poll`,
-                  IF(snmp_disable=1 OR status=0, 0, COALESCE(`last_discovered` <= DATE_ADD(DATE_ADD(NOW(), INTERVAL -%s SECOND), INTERVAL `last_discovered_timetaken` SECOND), 1)) AS `discover`
+                  IF (%s < `last_polled_timetaken` * 1.25, 0, COALESCE(`last_polled` <= DATE_ADD(DATE_ADD(NOW(), INTERVAL -%s SECOND), INTERVAL `last_polled_timetaken` SECOND), 1)) AS `poll`,
```

Comment:
I'm not sure this is exactly what you want for polling.  say last_polled_timetaken is the worst case (300s) and it just missed starting before the restart.  It will not poll until 375s after the service as started, skipping an interval.

I assume you are attempting to prevent double polling here.  However, the service already locks devices while they are being polled so they can't be double polled.

Reply:
Discovery is a bit different, the locks are much shorter than the discovery interval, so there is a chance for an early rediscover.

Reply:
True, the locks should be sufficient. I'll leave it in for discovery.